src/slices/catalogSlice.ts

```diff
@@ -0,0 +1,29 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { ProductModel } from '../models/ProductModel';
+import { getProductsThunkAction } from '../actions/getProductsThunkAction';
+
+export interface State {
+  products: {
+    [productId: number]: ProductModel;
+  };
+  productsIds: number[];
+}
+
+const initialState: State = {
+  products: {},
+  productsIds: [],
+};
+
+export const catalogSlice = createSlice({
+  name: 'catalog',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(getProductsThunkAction.fulfilled, (state, action) => {
+      state.productsIds = action.payload.map((product: any) => product.id);
```

Comment:
Do not use any if it's possible. You can set a type for the action: action: ActionPayload<YourType>, and then use it

Reply:
I use PayloadAction<ProductModel[]> as a type for my action in the extraReducer